### ISSUE TITLE
BBC - Test::Vars broken on 5.37.8

### ISSUE THREAD

**Issue by demerphq** [Perl/perl5#20723]
Module:
Test-Vars

**Description**
Test-Vars fails to build under blead. 

**Steps to Reproduce**
make test
```
t/07_stub_sub_bug.t ...... 1/? 
#   Failed test 'got expected output from test_vars'
#   at t/07_stub_sub_bug.t line 25.
#     Structures begin differing at:
#          $got->[0][2][1][1] = '$x is used once in &StubSub::foo '
#     $expected->[0][2][1][1] = '$x is used once in &StubSub::foo at t/lib/StubSub.pm line 13'
# Looks like you failed 1 test of 2.
t/07_stub_sub_bug.t ...... Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/2 subtests 
```

**Expected behavior**
No error.

**Perl configuration**

```
$ perl -V
Summary of my perl5 (revision 5 version 37 subversion 8) configuration:
  Commit id: f509a5163af6281fb894ec68237c699a4857fbe9
  Platform:
    osname=linux
    osvers=5.14.0-1054-oem
    archname=x86_64-linux-thread-multi
    uname='linux oncidium 5.14.0-1054-oem #61-ubuntu smp fri oct 14 13:05:50 utc 2022 x86_64 x86_64 x86_64 gnulinux '
    config_args='-de -Dcc=ccache gcc -Dld=gcc -Dprefix=/home/yorton/perl5/perlbrew/perls/latest_blead -Dman1dir=none -Dman3dir=none -Dusemultiplicity -Dusethreads -DDEBUGGING -Dusedevel -Aeval:scriptdir=/home/yorton/perl5/perlbrew/perls/latest_blead/bin'
    hint=recommended
    useposix=true
    d_sigaction=define
    useithreads=define
    usemultiplicity=define
    use64bitint=define
    use64bitall=define
    uselongdouble=undef
    usemymalloc=n
    default_inc_excludes_dot=define
  Compiler:
    cc='gcc'
    ccflags ='-D_REENTRANT -D_GNU_SOURCE -fwrapv -DDEBUGGING -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64'
    optimize='-O2 -g'
    cppflags='-D_REENTRANT -D_GNU_SOURCE -fwrapv -DDEBUGGING -fno-strict-aliasing -pipe -fstack-protector-strong -I/usr/local/include'
    ccversion=''
    gccversion='9.4.0'
    gccosandvers=''
    intsize=4
    longsize=8
    ptrsize=8
    doublesize=8
    byteorder=12345678
    doublekind=3
    d_longlong=define
    longlongsize=8
    d_longdbl=define
    longdblsize=16
    longdblkind=3
    ivtype='long'
    ivsize=8
    nvtype='double'
    nvsize=8
    Off_t='off_t'
    lseeksize=8
    alignbytes=8
    prototype=define
  Linker and Libraries:
    ld='gcc'
    ldflags =' -fstack-protector-strong -L/usr/local/lib'
    libpth=/usr/local/lib /usr/lib/x86_64-linux-gnu /usr/lib /usr/lib64
    libs=-lpthread -ldl -lm -lcrypt -lutil -lc
    perllibs=-lpthread -ldl -lm -lcrypt -lutil -lc
    libc=libc-2.31.so
    so=so
    useshrplib=false
    libperl=libperl.a
    gnulibc_version='2.31'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs
    dlext=so
    d_dlsymun=undef
    ccdlflags='-Wl,-E'
    cccdlflags='-fPIC'
    lddlflags='-shared -O2 -g -L/usr/local/lib -fstack-protector-strong'


Characteristics of this binary (from libperl): 
  Compile-time options:
    DEBUGGING
    HAS_TIMES
    MULTIPLICITY
    PERLIO_LAYERS
    PERL_COPY_ON_WRITE
    PERL_DONT_CREATE_GVSV
    PERL_HASH_FUNC_SIPHASH13
    PERL_HASH_USE_SBOX32
    PERL_MALLOC_WRAP
    PERL_OP_PARENT
    PERL_PRESERVE_IVUV
    PERL_TRACK_MEMPOOL
    PERL_USE_DEVEL
    PERL_USE_SAFE_PUTENV
    USE_64_BIT_ALL
    USE_64_BIT_INT
    USE_ITHREADS
    USE_LARGE_FILES
    USE_LOCALE
    USE_LOCALE_COLLATE
    USE_LOCALE_CTYPE
    USE_LOCALE_NUMERIC
    USE_LOCALE_TIME
    USE_PERLIO
    USE_PERL_ATOF
    USE_REENTRANT_API
    USE_THREAD_SAFE_LOCALE
  Built under linux
  Compiled at Jan 18 2023 14:03:23
  %ENV:
    PERLBREW_CONFIGURE_FLAGS="-de -Dcc=ccache\ gcc -Dld=gcc"
    PERLBREW_HOME="/home/yorton/.perlbrew"
    PERLBREW_MANPATH="/home/yorton/perl5/perlbrew/perls/latest_blead/man"
    PERLBREW_PATH="/home/yorton/perl5/perlbrew/bin:/home/yorton/perl5/perlbrew/perls/latest_blead/bin"
    PERLBREW_PERL="latest_blead"
    PERLBREW_ROOT="/home/yorton/perl5/perlbrew"
    PERLBREW_SHELLRC_VERSION="0.88"
    PERLBREW_VERSION="0.88"
  @INC:
    /home/yorton/perl5/perlbrew/perls/latest_blead/lib/site_perl/5.37.8/x86_64-linux-thread-multi
    /home/yorton/perl5/perlbrew/perls/latest_blead/lib/site_perl/5.37.8
    /home/yorton/perl5/perlbrew/perls/latest_blead/lib/5.37.8/x86_64-linux-thread-multi
    /home/yorton/perl5/perlbrew/perls/latest_blead/lib/5.37.8

```

**Comment by richardleach** [comment]
I think this is still a known breakage because of new OPs added in this development cycle. (Originally raised in https://github.com/Perl/perl5/issues/20114). A potential patch has been submitted upstream, but has stalled for want of someone who knows Test::Vars to assess its appropriateness: https://github.com/houseabsolute/p5-Test-Vars/issues/47

**Comment by jkeenan** [comment]
On 1/18/23 11:34, Richard Leach wrote:
> I think this is still a known breakage because of new OPs added in this 
> development cycle. (Originally raised in #20114 
> <https://github.com/Perl/perl5/issues/20114>). A potential patch has 
> been submitted upstream, but has stalled for want of someone who knows 
> Test::Vars to assess its appropriateness: houseabsolute/p5-Test-Vars#47 
> <https://github.com/houseabsolute/p5-Test-Vars/issues/47>
> 

Per https://metacpan.org/pod/Test::Vars, the maintainer of Test-Vars is 
looking for a replacement.  So unless one shows up soon there is likely 
to be a lot of CPAN breakage.

**Comment by jkeenan** [comment]
> On 1/18/23 11:34, Richard Leach wrote: I think this is still a known breakage because of new OPs added in this development cycle. (Originally raised in #20114 <#20114>). A potential patch has been submitted upstream, but has stalled for want of someone who knows Test::Vars to assess its appropriateness: [houseabsolute/p5-Test-Vars#47](https://github.com/houseabsolute/p5-Test-Vars/issues/47) <[houseabsolute/p5-Test-Vars#47](https://github.com/houseabsolute/p5-Test-Vars/issues/47)>
> Per https://metacpan.org/pod/Test::Vars, the maintainer of Test-Vars is looking for a replacement. So unless one shows up soon there is likely to be a lot of CPAN breakage.

Module has not been updated on CPAN, so breakage continues:
http://fast-matrix.cpantesters.org/?dist=Test-Vars

**Comment by jkeenan** [comment]
> > On 1/18/23 11:34, Richard Leach wrote: I think this is still a known breakage because of new OPs added in this development cycle. (Originally raised in #20114 <#20114>). A potential patch has been submitted upstream, but has stalled for want of someone who knows Test::Vars to assess its appropriateness: [houseabsolute/p5-Test-Vars#47](https://github.com/houseabsolute/p5-Test-Vars/issues/47) <[houseabsolute/p5-Test-Vars#47](https://github.com/houseabsolute/p5-Test-Vars/issues/47)>
> > Per https://metacpan.org/pod/Test::Vars, the maintainer of Test-Vars is looking for a replacement. So unless one shows up soon there is likely to be a lot of CPAN breakage.
> 
> Module has not been updated on CPAN, so breakage continues: http://fast-matrix.cpantesters.org/?dist=Test-Vars

Still broken in perl-5.41.1; see http://fast-matrix.cpantesters.org/?dist=Test-Vars;perl=5.41.1;reports=1

**Comment by richardleach** [comment]
> Module has not been updated on CPAN, so breakage continues:
> http://fast-matrix.cpantesters.org/?dist=Test-Vars

https://github.com/houseabsolute/p5-Test-Vars/pull/52 is the latest PR. It's been tested by @hakonhagland, but has stalled waiting on module maintainers to feedback or merge.

**Comment by jkeenan** [comment]
I assumed co-maint on Test-Vars, applied and tested @richardleach's patch, and released version 0.017 last month.  It's doing well on [CPANtesters](http://fast-matrix.cpantesters.org/?dist=Test-Vars).  Closing this ticket.
